src/test/java/org/prebid/server/bidder/adocean/AdoceanBidderTest.java

```diff
@@ -0,0 +1,380 @@
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+import static java.util.Arrays.asList;
+
+public class AdoceanBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "https://{{Host}}";
+
+    private AdoceanBidder adoceanBidder;
+
+    @Before
+    public void setUp() {
+        adoceanBidder = new AdoceanBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> new SmartrtbBidder("invalid_url", jacksonMapper))
+                .withMessage("URL supplied is not valid: invalid_url");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfImpressionListSizeIsZero() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(emptyList())
+                .build();
+
+        // when
+        final Result<List<HttpRequest<Void>>> result = adoceanBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput("No impression in the bid request"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorIfImpExtCouldNotBeParsed() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .id("123")
+                        .ext(mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()))));
+        // when
+        final Result<List<HttpRequest<Void>>> result = adoceanBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Cannot deserialize instance");
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSetReturnErrorIfRequestExists() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .user(User.builder()
+                        .ext(ExtUser.builder()
+                                .consent("COwK6gaOwK6gaFmAAAENAPCAAAAAAAAAAAAAAAAAAAAA."
+                                        + "IFoEUQQgAIQwgIwQABAEAAAAOIAACAIAAAAQAIAgEAACEAAAAAgAQBAAAAAAAGBAAgAAAAAA"
+                                        + "AFAAECAAAgAAQARAEQAAAAAJAAIAAgAAAYQEAAAQmAgBC3ZAYzUw").build())
+                        .build())
+                .imp(asList(Imp.builder()
+                                .id("ao-test")
+                                .banner(Banner.builder().format(singletonList(Format.builder().w(300).h(250).build()))
+                                        .id("banner_id").build()).ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpAdocean.of("myao.adocean.pl", "tmYF.DMl7ZBq.Nqt2Bq4FutQTJfTpxCOmtNPZoQUDcL.G7",
```

Comment:
Does Banner and User is really necessary for this test ?

Also use codeStyle like this
```
   Imp.builder()
                                .id("ao-test")
                                .ext(mapper.valueToTree(ExtPrebid.of(null,
                                        ExtImpAdocean.of("myao.adocean.pl", "tmYF.DMl7ZBq.Nqt2Bq4FutQTJfTpxCOmtNPZoQUDcL.G7",
                                                "adoceanmyaozpniqis"))))
                                .build()))
```